src/mainchain/MainchainGatewayV3.sol

```diff
@@ -353,6 +378,7 @@ contract MainchainGatewayV3 is
     address mainchainWeth = address(wrappedNativeToken);
 
     _request.info.validate();
+    _requireNotRestricted(msg.sig, _request.info.erc);
```

Comment:
This can be reached from fallback and the msg.sig is not restricted. Is it intentional?

Reply:
no I will delete that logic.